### ISSUE TITLE
Web Interface of UShER has issues?

### ISSUE THREAD

**Issue by bikudh** [cov-lineages/pango-designation#577]
I was analyzing the data for emerging mutations within BA.3 lineage. Ran the Usher but this is what I got, all of the BA.3 are missassigned in Usher.
![image](https://user-images.githubusercontent.com/88990329/164768905-6b5e14d3-a3f6-4584-afa2-bf51c58f0e17.png)

Link to Usher
 https://genome.ucsc.edu/cgi-bin/hgPhyloPlace?db=wuhCor1&phyloPlaceTree=hgPhyloPlaceData/wuhCor1/public.plusGisaid.latest.masked.pb&subtreeSize=1000&remoteFile=https%3A%2F%2Flapis.cov-spectrum.org%2Fgisaid%2Fv1%2Fsample%2Fgisaid-epi-isl%3ForderBy%3Drandom%26limit%3D1000%26dateFrom%3D2022-01-17%26aaMutations%3DORF1b%253AS959P%26pangoLineage%3DBA.3%26host%3DHuman 

https://nextstrain.org/fetch/genome.ucsc.edu/trash/ct/singleSubtreeAuspice_genome_2b434_2e22e0.json?c=pango_lineage 
This is likely the wrong forum to put this and please close the ticket after you see this so that it can be addressed. @AngieHinrichs 

Thank you!

**Comment by silcn** [comment]
Looks like more misassignment by pangoLEARN, lots of that going on at the moment. These aren't really BA.3 and Usher figures this out.

**Comment by AngieHinrichs** [comment]
TL;DR what @silcn said, but with many more words and some links to pictures.  😆 

Hi @bikudh, it looks like you're starting from a search on cov-spectrum.org in which you're specifying BA.3 with ORF1b:S959P, and then clicking through to the UShER web interface, is that correct?

I am obviously a biased source here, but -- I think the problem is that cov-spectrum.org is using pangoLEARN assignments (instead of the new pangolin v4 default, UShER) to determine which sequences are BA.3.  (@chaoran-chen how are you getting the assignments?  from GISAID?)   Lately pangoLEARN has been getting a lot of false positives for many Omicron sublineages.  In the latest UCSC/UShER tree (2022-04-20), there are 641 sequences on the branch for BA.3.  However, there are 1472 sequences in the tree that pangoLEARN assigns BA.3 (all 641 of the BA.3 branch sequences, 378 BA.2-branch sequences, and various other Omicron lineage and recombinant branches).  

You can use the UShER web interface's links to subtrees to evaluate the differences in lineage assignment for yourself.  For example, here is the subtree1 that I get when repeating your query, colored by "Pango lineage assigned by UShER":
https://nextstrain.org/fetch/hgwdev.gi.ucsc.edu/~angie/pango-designation-577.subtree1.json?c=pango_lineage_usher&label=nuc%20mutations:G670T
As you can see, all of subtree1 is in the BA.2.10 branch of the UShER tree (blue); uploaded sequences are yellow.  If you change the color to just "Pango lineage" (assigned by pangolin/pangoLEARN, I should update the label), there are quite a few more colors:
https://nextstrain.org/fetch/hgwdev.gi.ucsc.edu/~angie/pango-designation-577.subtree1.json?c=pango_lineage&label=nuc%20mutations:G670T
Overall, pangoLEARN assigns BA.2.10 to most of those closely related sequences, but with a smattering of other lineages throughout the subtree.  pangoLEARN sometimes assigns different lineages to sequences that are very similar to each other.  It's kind of a black box model, optimized for speed over precision which is why scorpio/constellations was added in pangolin v3 to check specific mutations for VOCs, and why usher is the default analysis mode in pangolin v4.

Since there are only 641 sequences on the BA.3 branch of the UShER tree at the moment, you can get a Nextstrain view of the entire branch using the web interface.  I went to https://genome.ucsc.edu/cgi-bin/hgPhyloPlace and pasted in the name of a designated BA.3 sequence (from the file lineages.csv in this repository), SouthAfrica/NICD-N22239/2021, set the subtree size to 1500, and clicked upload (and waited...).  Here is a link to the BA.3 branch, colored by genotype at ORF1ab:5360 which I think is the same as ORF1b:959: 
https://nextstrain.org/fetch/hgwdev.gi.ucsc.edu/~angie/pango-designation-577.BA.3.json?c=gt-ORF1ab_5360&label=nuc%20mutations:C832T,C9534T,C11235T,C26060T,A29510C

The UShER tree is far from perfect! but at least you can examine it in terms of the mutations associated with its nodes and sequences.

**Comment by chaoran-chen** [comment]
@AngieHinrichs, yes, indeed. We currently just take the values from GISAID.

**Comment by bikudh** [comment]
@AngieHinrichs 
Thank you for explaining in detail. It all makes sense and very help for me to navigate things in future. Your analysis is exactly what I was trying to do as I saw a cluster of sequences with the mutation.  cov-spectrum has shortcuts to quite a few things  and it is very handy to investigate questions. It looks like they are basically making the data from GISAID user friendly and easy to navigate.  I guess there are three different prediction tools for lineage and there is still discrepancy between them. I am sure people are working to fix the issue. I have seen flag in GISAID database on the discrepancy.

Again I am really thankful to you for describing everything in detail. I will close this as I do not want this to distract others.
